python.d/proxysql.chart.py

```diff
@@ -0,0 +1,349 @@
+        'lines': [
+        ]},
+    'commands_duration': {
+        'options': [None, 'ProxySQL Commands Duration', 'ms', 'commands', 'proxysql.commands_duration', 'line'],
+        'lines': [
+        ]}
+}
+
+
+class Service(MySQLService):
+    def __init__(self, configuration=None, name=None):
+        MySQLService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.queries = dict(global_status=QUERY_GLOBAL,
+                            connection_pool_status=QUERY_CONNECTION_POOL,
+                            commands_status=QUERY_COMMANDS)
+
+    def _get_data(self):
+        raw_data = self._get_raw_data(description=True)
+
+        if not raw_data:
+            return None
+
+        to_netdata = dict()
+
+        if 'global_status' in raw_data:
+            global_status = dict(raw_data['global_status'][0])
+            for key in global_status:
+                if key.lower() in GLOBAL_STATS:
+                    to_netdata[key.lower()] = global_status[key]
+
+        if 'connection_pool_status' in raw_data:
+
+            to_netdata['bytes_data_recv'] = 0
+            to_netdata['bytes_data_sent'] = 0
+
+            for record in raw_data['connection_pool_status'][0]:
+                backend = self._generate_backend(record)
+                name = self._generate_backend_name(backend)
+
+                for key in backend:
+                    if key in CONNECTION_POOL_STATS:
+                        if key == 'status':
+                            backend[key] = self._convert_status(backend[key])
+
+                            if len(self.charts) > 0:
+                                if (name + '_status') not in self.charts['pool_status']:
+                                    self._add_backend_dimensions(name)
+
+                        to_netdata[name + "_" + key] = backend[key]
+
+                    if key == 'bytes_data_recv':
+                        to_netdata['bytes_data_recv'] += int(backend[key])
+
+                    if key == 'bytes_data_sent':
+                        to_netdata['bytes_data_sent'] += int(backend[key])
+
+        if 'commands_status' in raw_data:
+            for record in raw_data['commands_status'][0]:
+                cmd = self._generate_command_stats(record)
+
+                if len(self.charts) > 0:
+                    if (cmd['name'] + '_count') not in self.charts['commands_count']:
+                        self._add_command_dimensions(cmd['name'])
+                        self._add_histogram_chart(cmd)
+
+                    to_netdata[cmd['name'] + '_count'] = cmd['count']
+                    to_netdata[cmd['name'] + '_duration'] = cmd['duration']
+                    for histogram in cmd['histogram']:
+                        dimId = 'commands_histogram_' + \
+                            cmd['name'] + '_' + histogram
```

Comment:
Please use `format` for a string concatenation if you have > 2 strings.
```python
dimId = 'commands_histogram_{1}_{2}'.format(cmd['name'], histogram)
```

Reply:
Done :)